### PR TITLE
Assessments widget pagination tests

### PR DESCRIPTION
**Pull request by Vyacheslau**

### CODE REVIEW

**Comment by egorhm** [comment]
please check compliance with linting rules

File: test/selenium/src/lib/rest/template_provider.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      obj = copy.deepcopy(TemplateProvider.parsed_data[obj_name])
+    except Exception:
+      path = os.path.join(os.path.dirname(__file__), TemplateProvider.RELATIVE_PATH_TEMPLATE.format(obj_name))
+      with open(path) as json_file:
+        json_data = json_file.read()
+      data = json.loads(json_data)
+      TemplateProvider.parsed_data[obj_name] = data
+      obj = copy.deepcopy(data)
+    for k in kwargs:
+      obj[k] = kwargs[k]
```
Comment (egorhm) [inline test/selenium/src/lib/rest/template_provider.py]: it's better to use the following approach:
for k, v in kwargs.iteritems():
    obj[k] = v

File: test/selenium/src/lib/rest/template_provider.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    result = dict()
+    result[obj_name] = obj
+    return result
```
Comment (egorhm) [inline test/selenium/src/lib/rest/template_provider.py]: it's better to use this approach
```python
obj.update(kwargs)
return {obj_name: obj}
```

File: test/selenium/src/lib/constants/url.py [hunk 3]
```diff
@@ -3,11 +3,14 @@
 # Created By: jernej@reciprocitylabs.com
 # Maintained By: jernej@reciprocitylabs.com
 
+API = "api"
+AUTHENTICATION = API + "/roles?name__in=Auditor&_=1463991908086"
```
Comment (alieh-rymasheuski) [inline test/selenium/src/lib/constants/url.py]: Is it OK to keep the session ID (or whatever `_` denotes) hard-coded?

File: test/selenium/src/lib/page/widget/generic_widget.py [hunk 4]
```diff
@@ -106,15 +106,22 @@ class Controls(Widget):
   _locator_widget = locator.WidgetBar.CONTROLS
   _locator_filter = locator.WidgetControls
 
-  def __init__(self, driver,):
+  def __init__(self, driver, ):
```
Comment (alieh-rymasheuski) [inline test/selenium/src/lib/page/widget/generic_widget.py]: The comma is not needed too.

File: test/selenium/src/lib/rest/service.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+import requests
+
+from lib import environment
+from lib.constants import url
+from lib.rest.template_provider import generate_objects_list
+
+
+class BusinessObject(object):
+  """Model representing business entities used for REST queries."""
```
Comment (alieh-rymasheuski) [inline test/selenium/src/lib/rest/service.py]: Please use the same docstring format in all your code (it seems to me we have a convention not to put the period in the end of one-line docstrings, just like you did in other modules).

File: test/selenium/src/lib/rest/service.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+  def create_objects(self, obj_type, number=1, title_postfix=None, **kwargs):
+    """The method sends HTTP request for objects creation via REST API."""
+
+    def get_list_of_created_objects(response):
+      """The function forms the list of business entities from response."""
+      business_objects_list = list()
+      for object_element in json.loads(response.text):
+        object_key = object_element[1].keys()[0]
+        created_object = object_element[1][object_key]
```
Comment (alieh-rymasheuski) [inline test/selenium/src/lib/rest/service.py]: You can do this like `object_key, created_object = object_element.pop()`.

Comment (alieh-rymasheuski) [inline test/selenium/src/lib/rest/service.py]: `pop` is a destructive operation (it removes the key-value pair that is returned from the dict), but it should not be a problem as you don't reuse the dict.

File: test/selenium/src/lib/rest/template_provider.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+      with open(path) as json_file:
+        json_data = json_file.read()
+      data = json.loads(json_data)
+      TemplateProvider.parsed_data[obj_type] = data
+      obj = copy.deepcopy(data)
+    obj.update(kwargs)
+    return {obj_type: obj}
+
+
+def generate_objects_list(number, object_type, title_postfix=None, **kwargs):
```
Comment (alieh-rymasheuski) [inline test/selenium/src/lib/rest/template_provider.py]: I'd change the parameter list to `(obj_type, number, title_postfix=None, **kwargs)` like parameter lists of your other functions.

File: test/selenium/src/lib/rest/service.py [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+                                         created_object["selfLink"],
+                                         created_object["type"])
+        business_objects_list.append(business_object.as_dict())
+      return business_objects_list
+
+    objects = generate_objects_list(number, obj_type,
+                                    title_postfix=title_postfix, **kwargs)
+
+    request_body = json.dumps(objects)
+    headers = self.BASIC_HEADERS
```
Comment (alieh-rymasheuski) [inline test/selenium/src/lib/rest/service.py]: Please use `headers = self.BASIC_HEADERS.copy()` as `dict` is a reference type.

File: test/selenium/src/lib/rest/template/assessment.json [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+  "design": null,
+  "operationally": null,
+  "context": null,
+  "recipients": ",,",
+  "program": {
+    "id": 11,
+    "href": "/api/programs/11",
+    "type": "Program"
+  }
+}
```
Comment (alieh-rymasheuski) [inline test/selenium/src/lib/rest/template/assessment.json]: Please end all your files with newlines.

File: test/selenium/src/tests/conftest.py [hunk 10]
```diff
@@ -5,11 +5,13 @@
 
 """PyTest fixtures"""
 
-import pytest   # pylint: disable=import-error
+import pytest  # pylint: disable=import-error
 
 from lib import constants
-from lib.page.widget import info_widget
+from lib.business_layer.rest import ProgramsService, AuditsService, \
```
Comment (alieh-rymasheuski) [inline test/selenium/src/tests/conftest.py]: Please avoid backslashes in line endings (use [parenthesed imports](https://www.python.org/dev/peps/pep-0328/#rationale-for-parentheses)).

File: test/selenium/src/tests/conftest.py [hunk 11]
```diff
@@ -71,8 +73,8 @@ def new_program(selenium, new_control):
   saved modal"""
   # pylint: disable=redefined-outer-name
   modal = conftest_utils.get_lhn_accordion(
-      selenium, constants.element.Lhn.PROGRAMS)\
-      .create_new()
+    selenium, constants.element.Lhn.PROGRAMS) \
+    .create_new()
```
Comment (alieh-rymasheuski) [inline test/selenium/src/tests/conftest.py]: We can move `.create_new()` to the previous line and thus get rid of `\`.

File: test/selenium/src/tests/test_audit_page.py [hunk 12]
```diff
@@ -0,0 +1,29 @@
+# Copyright (C) 2016 Google Inc., authors, and contributors <see AUTHORS file>
+# Licensed under http://www.apache.org/licenses/LICENSE-2.0 <see LICENSE file>
+# Created By: Viachaslau_Karachun@epam.com
+# Maintained By: Viachaslau_Karachun@epam.com
+
+"""Tests Audit object"""
+# pylint: disable=no-self-use
+# pylint: disable=invalid-name
+# pylint: disable=too-few-public-methods
+# pylint: disable=unused-argument
```
Comment (alieh-rymasheuski) [inline test/selenium/src/tests/test_audit_page.py]: Don't forget to remove the unneeded `pylint disable` statements when the tests are added. :)

**Pull request closed without merging**